tests/providers/slack/chat-message.ts

```diff
@@ -0,0 +1,195 @@
+      handler.setConfig(handlerConfig);
+
+      try {
+        // Set up initial sync position
+        const syncPosition: SyncHandlerPosition = {
+          _id: `${providerId}-${handlerName}`,
+          providerId,
+          handlerId: handler.getId(),
+          accountId: provider.getAccountId(),
+          status: SyncHandlerStatus.ENABLED,
+        };
+
+        // Start the sync process
+        const response = await handler._sync(api, syncPosition);
+        const results = <SchemaRecord[]>response.results;
+
+        // Extract chat groups and messages from the results
+        const chatGroups = <SchemaSocialChatGroup[]>results.filter(result => result.schema === CONFIG.verida.schemas.CHAT_GROUP);
+        const chatMessages = <SchemaSocialChatMessage[]>results.filter(result => result.schema === CONFIG.verida.schemas.CHAT_MESSAGE);
+
+        // Ensure results are returned
+        assert.ok(results && results.length, "Have results returned");
+
+        // Check IDs in the returned items
+        CommonTests.checkItem(results[0], handler, provider);
+
+        // Verify sync status is active
+        assert.equal(
+          SyncHandlerStatus.SYNCING,
+          response.position.status,
+          "Sync is active"
+        );
+
+        // Ensure the message batch per group works
+        const firstGroupId = chatMessages[0].groupId;
+        const firstGroupMessages = chatMessages.filter(msg => msg.groupId === firstGroupId);
+        assert.equal(firstGroupMessages.length, handlerConfig.messagesPerGroupLimit, "Processed correct number of messages per group");
+
+         /**
+          * Start the second sync batch process
+          */
+        const secondBatchResponse = await handler._sync(api, response.position);
+        const secondBatchResults = <SchemaRecord[]>secondBatchResponse.results;
+
+        // Extract chat groups and messages from the second batch results
+        const secondBatchChatGroups = <SchemaSocialChatGroup[]>secondBatchResults.filter(result => result.schema === CONFIG.verida.schemas.CHAT_GROUP);
+        const secondBatchChatMessages = <SchemaSocialChatMessage[]>secondBatchResults.filter(result => result.schema === CONFIG.verida.schemas.CHAT_MESSAGE);
+
+        // Ensure second batch results are returned
+        assert.ok(secondBatchResults && secondBatchResults.length, "Have second batch results returned");
+
+        // Check IDs in the returned items for the second batch
+        CommonTests.checkItem(secondBatchResults[0], handler, provider);
+
+        // Verify sync status is still active for the second batch
+        assert.equal(
+          SyncHandlerStatus.SYNCING,
+          secondBatchResponse.position.status,
+          "Sync is still active after second batch"
+        );
+
+        // Check if synced every chat group correctly
+        const syncedGroup = (secondBatchChatGroups.filter(group => group.sourceId === secondBatchChatMessages[0].groupId))[0];
```

Comment:
This only checks one group, not all the groups

Reply:
> This only checks one group, not all the groups

Done